crates/networks/src/lib.rs

```diff
@@ -92,6 +92,8 @@ impl Networks {
             return Err(Error::AlreadyExists);
         }
 
+        network.internal_id = Some(self.networks.len() as u32);
```

Comment:
what about networks that are deserialized directly from `networks.json`? we need to find a way to set the ID for them too when initializing

Reply:
unless I'm missing, if the network is already in the `networks.json` that means it was added by this code and already has an `internal_id` field set

or is this to support adding a new network by manually editing the `networks.json` file?